servicenow/test_servicenow.py

```diff
@@ -0,0 +1,217 @@
+# (C) Datadog, Inc. 2010-2016
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+
+# stdlib
+from nose.plugins.attrib import attr
+
+# 3p
+import mock
+import json
+
+# project
+from tests.checks.common import AgentCheckTest
+from checks import AgentCheck, CheckException
+
+
+instance = {
+    'url': "https://dev6047.service-now.com",
+    'basic_auth': {'user': 'admin', 'password': 'Service@123'},
+    'default_batch_size': 100
+}
+
+CONFIG = {
+    'init_config': {'default_timeout': 10, 'min_collection_interval': 5},
+    'instances': [
+        {
+            'url': "https://dev60479.service-now.com",
+            'basic_auth': {'user': 'admin', 'password': 'Service@12'},
+            'default_batch_size': 100
+        }
+    ]
+}
+
+
+def mock__process_and_cache_relation_types(params):
+    return
+
+
+def mock__process_components(params):
+    return
+
+
+def mock__process_component_relations(params):
+    return
+
+
+def mock_collect_components():
+    """
+    Mock behaviour(response) from ServiceNow API for Components(CIs)
+    """
+    response = {'result': [{'sys_class_name': 'cmdb_ci_computer', 'sys_id': '00a96c0d3790200044e0bfc8bcbe5db4',
+                            'sys_created_on': '2012-02-18 08:14:21', 'name': 'MacBook Pro 15'}]}
+    return json.dumps(response)
+
+
+def mock_relation_types():
+    """
+    Mock behaviour for relation types
+    """
+    response = {'result': [{'parent_descriptor': 'Cools', 'sys_id': '53979c53c0a801640116ad2044643fb2'}]}
+    return json.dumps(response)
+
+
+def mock_relation_components():
+    """
+    Mock response from ServiceNow API for relation between components
+    """
+    response = {'result': [
+        {'type': {'link': 'https://dev60476.service-now.com/api/now/table/cmdb_rel_type/1a9cb166f1571100a92eb60da2bce5c5',
+                  'value': '1a9cb166f1571100a92eb60da2bce5c5'},
+         'parent': {'link': 'https://dev60476.service-now.com/api/now/table/cmdb_ci/451047c6c0a8016400de0ae6df9b9d76',
+                    'value': '451047c6c0a8016400de0ae6df9b9d76'},
+         'child': {'link': 'https://dev60476.service-now.com/api/now/table/cmdb_ci/53979c53c0a801640116ad2044643fb2',
+                   'value': '53979c53c0a801640116ad2044643fb2'}}
+    ]}
+    return json.dumps(response)
+
+# NOTE: Feel free to declare multiple test classes if needed
+
+
+@attr(requires='servicenow')
```

Comment:
Due to this the tests do not run in CI, so you want to drop this to make the tests run as part of the default unit tests